TUTORIAL.md

```diff
@@ -0,0 +1,245 @@
+
+###Step 3: Set up your PostsContrller
+
+### PostsController
+
+```ruby
+def new
+  @post = Post.new
+  @tag = @post.tags.build
+end
+```
+
+Our controller is going to render our form, but what is this `build` method? Our `build` method was given to use via our model associations, which we will take a look at later. It creates a new object in memory, so that our view can use it to display something. In our case, since we are using `fields_for`, it will display fields for our tags! If you delete `@tag = @post.tags.build` from your controller, your fields would disappear.
+
+Ok, so we can build our form dynamically, but how does Rails know where to submit it to? Luckily for us, Rails is really smart. When we passed the `@post` object to `form_for`, it was able to check for an instance of that post in the database. Since it was a new instance, by convention it decided that we are trying to create a new post and should be submitted to `/posts` via a `post` method. Let's take a look at the HTML that was generated for us by `form_for`.
+
+####Generated HTML From Ruby
+
+```html
+
+<form class="new_post" id="new_post" action="/posts" accept-charset="UTF-8" method="post">
+<input name="utf8" type="hidden" value="&#x2713;" />
+<input type="hidden" name="authenticity_token" value="IRnbjg0Lr7XuZmk1wqEewWqPbOvU0j5ymcne3Rq3MV3yHzhC1VbZsySOGggFVgM1zwL6b+E2+nGsnaQCQQfu6g==" />
+
+  <div class="field">
+    <label for="post_name">Name</label><br>
+    <input type="text" name="post[name]" id="post_name" /><br>
+    <label for="post_content">Content</label><br>
+    <textarea name="post[content]" id="post_content">
+    </textarea><br>
+    Tags:
+  </div>
+
+  <div class="field">
+    <label for="post_tags_attributes_0_name">Name</label>
+    <input type="text" name="post[tags_attributes][0][name]" id="post_tags_attributes_0_name" />
+  
+    <label for="post_tags_attributes_1_name">Name</label>
+    <input type="text" name="post[tags_attributes][1][name]" id="post_tags_attributes_1_name" />
+  
+    <label for="post_tags_attributes_2_name">Name</label>
+    <input type="text" name="post[tags_attributes][2][name]" id="post_tags_attributes_2_name" />
+  </div>
+
+  <div class="actions">
+    <input type="submit" name="commit" value="Create Post" />
+  </div>
+
+</form>
+```
+
+In the first section of our form, we can see our form helper generated `post[name]` and `post[content]`. This is great, we have access to our post name and content.
+
+#### `fields_for`
+Let's focus on our tag fields for a second, how did Rails know to create such a nicely nested structure?
+
+If we look at our form, we will see `<%= f.fields_for(:tags) do |tag_form| %>`. Our `fields_for` accepts a `:tags` object and by convention introspects into our `Post` model to see if it has a way to accept this new object. It is specifically looking for a macro called `accepts_nested_attributes_for :tags`. If it finds it, it will generate the nicely nested form that we see above.
```

Comment:
It doesn't introspect on Post because of convention, it introspects because the `f` `FormBuilder` object already knows about Post from your `form_for`